### PR TITLE
MI sample projects documentation for VS Code extension

### PR DESCRIPTION
**Pull request by ChinthakaJ98**
This PR will introduce documentations on how to deploy and work with MI sample projects via the MI VS Code Extension.

### CODE REVIEW

File: en/docs/learn/vscode-plugin-samples/database-polling.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Database Polling Sample
+
+This sample demonstrates database polling capability in the Micro Integrator.
+
+Four modules get created with this sample which are; a scheduled task called `DoctorsRecordsSyncTask`, a sequence called `DoctorsRecordsSyncSeq`, an endpoint called `DoctorsDataServiceEP` and a data service called `DoctorsDataService`.
+
+The data service module expose three database operations; `ReadDoctors`, `UpdateDoctorsSyncStatus` and `ReadDoctorsNotSync`.
```
Comment (DinithiDiaz) [inline en/docs/learn/vscode-plugin-samples/database-polling.md:7]:
```suggestion
The data service module exposes three database operations; `ReadDoctors`, `UpdateDoctorsSyncStatus`, and `ReadDoctorsNotSync`.
```

File: en/docs/learn/vscode-plugin-samples/database-polling.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+1. Create a database and a user with the following details and grant that user access to the database.
+
+    ```bash
+    CREATE DATABASE hospital_db;
+    USE hospital_db;
+    CREATE USER 'user'@'localhost' IDENTIFIED BY 'password';
+    GRANT ALL PRIVILEGES ON hospital_db.* TO 'user'@'localhost';
+    ```
+
+2. Create table and insert some records.
```
Comment (DinithiDiaz) [inline en/docs/learn/vscode-plugin-samples/database-polling.md:22]:
```suggestion
2. Create a table and insert some records.
```

File: en/docs/learn/vscode-plugin-samples/exception-handling.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# Exception Handling Sample
+
+This sample demonstrates the exception handling capabilities of the Micro Integrator.
+
+This sample consists of an API called `TimeoutAPI`, an Endpoint called `DelayHttpEP` and a Sequence called `TimeoutFailureSeq`
```
Comment (DinithiDiaz) [inline en/docs/learn/vscode-plugin-samples/exception-handling.md:5]:
```suggestion
This sample consists of an API called `TimeoutAPI`, an endpoint called `DelayHttpEP`, and a sequence called `TimeoutFailureSeq`.
```

File: en/docs/learn/vscode-plugin-samples/file-transfer.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# File Transfer Sample
+
+This sample demonstrates some file handling capabilities of the Micro Integrator.
+
+This sample contains an Inbound Endpoint called `StudentDataFileProcessInboundEP` which polls a particular location for available files. Also, it contains two sequences called `StudentDataFileProcessSeq` and `StudentDataFileErrorSeq`.
+
+The file inbound endpoint is listening on file path specified in the `File URI` parameter. The `transport.vfs.FileNamePattern` represent the file pattern to process. Once the inbound endpoint read a file successfully, its content is injected to the sequence for further mediation. In this sample, `StudentDataFileProcessSeq` logs the content of the received message.
+
+Once file reading is successful, the file will be moved to a location defined in the `Move After Process` parameter. If an error occurred while reading, then the file will be moved to the location specified in the transport.vfs.MoveAfterFailure parameter.
```
Comment (DinithiDiaz) [inline en/docs/learn/vscode-plugin-samples/file-transfer.md:9]:
```suggestion
Once file reading is successful, the file will be moved to a location defined in the `Move After Process` parameter. If an error occurred while reading, then the file will be moved to the location specified in the `transport.vfs.MoveAfterFailure` parameter.
```

File: en/docs/learn/vscode-plugin-samples/rest-data-service.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+1. Create a database and a user with the following details and grant that user access to the database.
+
+    ```bash
+    CREATE DATABASE school_db;
+    USE school_db;
+    CREATE USER 'user'@'localhost' IDENTIFIED BY 'password';
+    GRANT ALL PRIVILEGES ON school_db.* TO 'user'@'localhost';
+    ```
+
+2. Create table and insert some records.
```
Comment (ChinthakaJ98) [inline en/docs/learn/vscode-plugin-samples/rest-data-service.md:18]:
```suggestion
2. Create a table and insert some records.
```

File: en/docs/learn/vscode-plugin-samples/students-data-service.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+1. Create a database and a user with the following details and grant that user access to the database.
+
+    ```bash
+    CREATE DATABASE school_db;
+    USE school_db;
+    CREATE USER 'user'@'localhost' IDENTIFIED BY 'password';
+    GRANT ALL PRIVILEGES ON school_db.* TO 'user'@'localhost';
+    ```
+
+2. Create table and insert some records.
```
Comment (ChinthakaJ98) [inline en/docs/learn/vscode-plugin-samples/students-data-service.md:18]:
```suggestion
2. Create a table and insert some records.
```

File: en/docs/learn/vscode-plugin-samples/database-polling.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    ('jimmy carter', 'willow gardens general hospital', 'ent', '9.00 a.m - 11.00 a.m', 2500),
+    ('peter simon', 'clemency medical center', 'cardiology', '9.00 a.m - 11.00 a.m', 6000);
+    ```
+
+## Deploying the sample
+
+1. Open the sample by clicking on the **Database Polling** card.
+2. Give a folder location to save the sample.
+3. Open the `DoctorsDataService` data service artifact and update the RDBMS datasource in it by replacing the `URL`, `Username` and `Password` values with your database credentials.
+4. Download the `mysql-connector-j-8.0.32.jar` and add it to the `<PROJECT_WORKSPACE>/deployment/libs` folder.
```
Comment (DinithiDiaz) [inline en/docs/learn/vscode-plugin-samples/database-polling.md:61]: Let's add the download link for the JAR file here.

File: en/docs/learn/vscode-plugin-samples/rest-data-service.md [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    ('Tim', 'Summer School', 7),
+    ('Peter', 'Burnley High School', 10);
+    ```
+
+## Deploying the sample
+
+1. Open the sample by clicking on the **REST Data Service** card.
+2. Give a folder location to save the sample.
+3. Open the `RESTDataService` data service artifact and update the RDBMS datasource in it by replacing the `URL`, `Username` and `Password` values with your database credentials.
+4. Download the `mysql-connector-j-8.0.32.jar` and add it to the `<PROJECT_WORKSPACE>/deployment/libs` folder.
```
Comment (DinithiDiaz) [inline en/docs/learn/vscode-plugin-samples/rest-data-service.md:41]: Let's add the download link for the JAR file here.

File: en/docs/learn/vscode-plugin-samples/students-data-service.md [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+    ('Tim', 'Summer School', 7),
+    ('Peter', 'Burnley High School', 10);
+    ```
+
+## Deploying the sample
+
+1. Open the sample by clicking on the **Students Data Service** card.
+2. Give a folder location to save the sample.
+3. Open the `StudentDataService` data service artifact and update the RDBMS datasource in it by replacing the `URL`, `Username` and `Password` values with your database credentials.
+4. Download the `mysql-connector-j-8.0.32.jar` and add it to the `<PROJECT_WORKSPACE>/deployment/libs` folder.
```
Comment (DinithiDiaz) [inline en/docs/learn/vscode-plugin-samples/students-data-service.md:41]: Let's add the download link for the JAR file here.

File: en/docs/learn/vscode-plugin-samples/database-polling.md (hunk 7 above)
Comment (DinithiDiaz) [inline en/docs/learn/vscode-plugin-samples/database-polling.md:61]:
```suggestion
4. Download the `mysql-connector-j-8.0.32.jar` and add it to the `<PROJECT_NAME>/deployment/libs` folder.
```

File: en/docs/learn/vscode-plugin-samples/rest-data-service.md (hunk 8 above)
Comment (DinithiDiaz) [inline en/docs/learn/vscode-plugin-samples/rest-data-service.md:41]:
```suggestion
4. Download the `mysql-connector-j-8.0.32.jar` and add it to the `<PROJECT_NAME>/deployment/libs` folder.
```

File: en/docs/learn/vscode-plugin-samples/students-data-service.md (hunk 9 above)
Comment (DinithiDiaz) [inline en/docs/learn/vscode-plugin-samples/students-data-service.md:41]:
```suggestion
4. Download the `mysql-connector-j-8.0.32.jar` and add it to the `<PROJECT_NAME>/deployment/libs` folder.
```

**Review by DinithiDiaz — DISMISSED** [review]

File: en/docs/learn/vscode-plugin-samples/database-polling.md (hunk 7 above)
Comment (ChinthakaJ98) [reply en/docs/learn/vscode-plugin-samples/database-polling.md]: Fixed in 8844ccd775aa7132437584a8e0baee0ad4e5982a

File: en/docs/learn/vscode-plugin-samples/rest-data-service.md (hunk 8 above)
Comment (ChinthakaJ98) [reply en/docs/learn/vscode-plugin-samples/rest-data-service.md]: Fixed in 8844ccd775aa7132437584a8e0baee0ad4e5982a

File: en/docs/learn/vscode-plugin-samples/students-data-service.md (hunk 9 above)
Comment (ChinthakaJ98) [reply en/docs/learn/vscode-plugin-samples/students-data-service.md]: Fixed in 8844ccd775aa7132437584a8e0baee0ad4e5982a

**Review by DinithiDiaz — APPROVED** [review]

**Pull request merged**